test/OpenTelemetry.Tests/Metrics/MetricExporterTests.cs

```diff
@@ -99,5 +152,35 @@ public override ExportResult Export(in Batch<Metric> batch)
                 return ExportResult.Success;
             }
         }
+
+        private class UpdateMetricPointExporter : BaseExporter<Metric>
+        {
+            private static int invocationCount = 1;
+            private readonly List<Metric> exportedItems;
+
+            public UpdateMetricPointExporter(List<Metric> exportedItems)
+            {
+                this.exportedItems = exportedItems;
+            }
+
+            public override ExportResult Export(in Batch<Metric> batch)
+            {
+                foreach (var metric in batch)
+                {
+                    if (invocationCount > 1)
+                    {
+                        foreach (ref var metricPoint in metric.GetMetricPoints())
+                        {
+                            metricPoint = default;
```

Comment:
This is allowed as of today.

Reply:
Bummer... good catch.

Reply:
Good call @utpilla! If you change this line...

https://github.com/open-telemetry/opentelemetry-dotnet/blob/5d9083b9739889e1642e7688832eb7d47f9141f7/src/OpenTelemetry/Metrics/MetricPointsAccessor.cs#L79

...to...

```csharp
            public readonly ref MetricPoint Current
```

...does that fix it?

Reply:
Yeah `ref readonly` should fix it. We would have to update the exporters though which should be fine.